core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -67,6 +67,15 @@ class KafkaZkClient private (zooKeeperClient: ZooKeeperClient, isSecure: Boolean
   // Only for testing
   private[kafka] def currentZooKeeper: ZooKeeper = zooKeeperClient.currentZooKeeper
 
+  /*
+  * This variable holds the Zookeeper session id at the moment the ZooKeeperClient object
+  * is created and the subsequent updates of the session id, it is possible that the session id change
+  * over the time for 'Session expired', for instance.
+  * */
+  private var previousZooKeeperSessionId = zooKeeperClient.sessionId
```

Comment:
The following is the comment from ZooKeeper.getSessionId(). Since ZK connects to the server asynchronously, the sessionId is not guaranteed to be available until the session is established.

```
    /**
     * The session id for this ZooKeeper client instance. The value returned is
     * not valid until the client connects to a server and may change after a
     * re-connect.
     *
     * This method is NOT thread safe
     *
     * @return current session id
     */
    public long getSessionId() {

```

Reply:
Oh, this could be a problem at the moment the session Id is collected for the first time. I will check where is the best place to assign it.
Thanks for the heads up.